setup.py

```diff
@@ -43,13 +43,30 @@
 
     BOKEHJS_ACTION="install" pip install .
 '''
-from setuptools import find_packages, setup
+from setuptools import find_namespace_packages, setup
+
+import sys
+from pathlib import Path
+
+ROOT = Path(__file__).resolve().parent
+sys.path.append(str(ROOT))  # avoid depending on `build_meta:__legacy__`
```

Comment:
`setuptools.build_meta:__legacy__` inserts the project directory as the first entry in `sys.path` (i.e. it replicates the behaviour of `python setup.py`), this allows importing local files, but also can generate some problems.

`setuptools.build_meta` does not add the local path to `sys.path`, it is up to the user to do it.

Reply:
At this point enough code has been deleted that I would probably just move everything back into `setup.py` and delete `_setup_support.py` altogether.

Reply:
Probably `sys.path.append` would still be required to import versioneer.

Reply:
Have you ever worked with [`setuptools-scm`](https://pypi.org/project/setuptools-scm/)? It is not built-in, but quite handy to use git tags as single source of truth for the version.

I use something similar to the following in all my projects:

```toml
# pyproject.toml
[build-system]
requires = ["setuptools>62.3.3", "setuptools_scm>=6.2"]
```

Reply:
I'd never heard of it until now, I will take a look